### DISCUSSION TITLE
IR Watchdog : Initial Tech Specs

### DISCUSSION THREAD

**Discussion by nelsonii** [ATMakersOrg/EyeAlert#12] [Ideas]
"It's a smoke alarm for IR" -- this is NOT a medical device NOR a redundant solution. The idea is to alert caregivers when "the eye gaze accidently gets unplugged" etc. It is NOT an eye tracker. 

**General Requirements**

1. Repeatable (easy to source parts).
2. Inexpensive (<$100).
3. Easy to use (no fancy UI).

**Initial Tech Thoughts**

1. Use the IR board that Chris and Bill have developed, with a QT Py as the microcontroller, for prototype.
2. The design will be such that it can be re-created WITHOUT any custom PCBs -- for folks who can't get them. Should be point-to-point wiring friendly.
3. It will be powered by USB. I think having a battery backup would be nice to have, but high on the list.
4. Use a different/custom board for the actual device? I really want battery backup, so we may want to go with boards that have that on-board.
5. Device will look for IR (there are questions about this in the Issues list). When IR goes away, provide audio and visual alert. I'm thinking buzzer/beep and a bright LED to start with.

**Future Tech Thoughts (Cool Improvements)**

1. Use a microcontroller which supports WiFi and have it send alerts out.

**Comment by ImproveAbility** [comment]
I think this is an intriguing idea.  There was a period of time where we were having some relatively consistent issues with cameras disconnecting.  This device would work in that situation.  I am currently dealing with an issue with Grid Pads right now that does have camera disconnect, so this is not a device that would detect all issues with an eye gaze system, but it is something that would be super useful.

**Reply by nelsonii** [reply]
Yep. Another tool for our toolbox. I'm thinking primary use is "eye gaze system turned off" (because, it *might* lock up, but still transmit IR).

This is a completely passive sensor -- it emits nothing, so it won't mess with equipment.

**Comment by ImproveAbility** [comment]
Question - how does the receiver "see" the IR - what kind of sensitivity does it have?  Where would it have to be positioned in order to pick up the IR from the camera?

**Reply by nelsonii** [reply]
There is a photodiode that looks for IR. It's a line-of-sight sensor which picks up IR, but doesn't try to "decode" it. 

This waveform image was measured from across the room, pointing a TV remote at it. (Tobii/etc would use similar illumination.)

![2022-05-05_21-06-18](https://user-images.githubusercontent.com/45170920/167059266-b25e59bd-995e-4fb7-ac8b-cefe8864a6a2.jpg)
 
The device can be mounted behind the user, with a view of the eye gaze -- the sensitivity is just like you get with a TV remote.
